### PR TITLE
Changelog for qcodes 0.29.0

### PR DESCRIPTION
**Pull request by astafan8**

### CODE REVIEW

File: docs/changes/0.29.0.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  moving the measurement arm. (#2626)
+
+
+---------
+Improved:
+---------
+
+- Documentation updates, contrib (#3375, #3370)
+- Fix shapes of dond datasets with additional set-points. (#3305)
+- Allow load experiment functions to accept duplicate experiment name
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:26]: This is available from  v0.28.0

File: docs/changes/0.29.0.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  and new experiment raises warning if duplicate experiment is created. (#3296)
+- Improvements to logging where we log the versions of installed packages in
+  the environment where QCoDeS runs. (#3289)
+- Enables dond to take inner actions that are passed in the Sweep instances.
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:30]: same

File: docs/changes/0.29.0.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  Actions for a sweep parameter will be called when the set value of that
+  parameter is changed. (#3283)
+- Raise ``ValueError`` if given more than one result for the same parameter name. (#3265)
+- Sphinx extension that documents parameter attributes on instrument classes.
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:34]: same

File: docs/changes/0.29.0.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  There are documented using a proxy object that contains a __repr__ that contains
+  the code to construct the parameter class. (#3224)
+
+
+-----------------
+Improved drivers:
+-----------------
+
+- Keithley 2600 SMU support to return both voltage and current compliance limit errors. (#3251, #3339, #3362)
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:43]: same for #3251

File: docs/changes/0.29.0.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+---------------
+
+- Add type packages discovered by mypy --install-types. (#3379)
+- ISSUE_TEMPLATE.md updates for reporting a bug. (#3345)
+- Resolved uncertainties package issues. (#3334)
+- Make it possible to manually trigger codeql. (#3329)
+- Logging update to explicitly set encoding for file handler. (#3327)
+- Create codeql-analysis.yml recommended by github. (#3314)
+- Remove redundant filters from test runner. (#3295)
+- Explicitly depend on setuptools. (#3294)
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:58]: same

File: docs/changes/0.29.0.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+- Remove precommit job that is now handled by dedicated precommit ci. (#3293)
+- Make qcodes install mostly pep517/518 compliant. (#3288)
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:60]: same

File: docs/changes/0.29.0.rst [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+- Add type packages discovered by mypy --install-types. (#3379)
+- ISSUE_TEMPLATE.md updates for reporting a bug. (#3345)
+- Resolved uncertainties package issues. (#3334)
+- Make it possible to manually trigger codeql. (#3329)
+- Logging update to explicitly set encoding for file handler. (#3327)
+- Create codeql-analysis.yml recommended by github. (#3314)
+- Remove redundant filters from test runner. (#3295)
+- Explicitly depend on setuptools. (#3294)
+- Remove precommit job that is now handled by dedicated precommit ci. (#3293)
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:59]: same

File: docs/changes/0.29.0.rst [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+- Make qcodes install mostly pep517/518 compliant. (#3288)
+- Work around legacy test issue with hf51.12.1 on conda forge. (#3278)
```
Comment (FarBo) [inline docs/changes/0.29.0.rst:61]: same

File: docs/changes/0.29.0.rst (hunk 1 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

File: docs/changes/0.29.0.rst (hunk 2 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

File: docs/changes/0.29.0.rst (hunk 4 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

File: docs/changes/0.29.0.rst (hunk 3 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

File: docs/changes/0.29.0.rst (hunk 5 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

File: docs/changes/0.29.0.rst (hunk 6 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

File: docs/changes/0.29.0.rst (hunk 8 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

File: docs/changes/0.29.0.rst (hunk 7 above)
Comment (astafan8) [reply docs/changes/0.29.0.rst]: thank you! removed

**Review by FarBo — APPROVED** [review]

**Pull request merged**